streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -488,7 +493,24 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
           registerReceiver(streamId, typ, host, executorId, receiverEndpoint, context.senderAddress)
         context.reply(successful)
       case AddBlock(receivedBlockInfo) =>
-        context.reply(addBlock(receivedBlockInfo))
+        if (WriteAheadLogUtils.isBatchingEnabled(ssc.conf, isDriver = true)) {
+          val f = Future {
+            if (active) {
+              addBlock(receivedBlockInfo)
+            } else {
+              throw new IllegalStateException("Receiver Tracker Endpoint shut down.")
+            }
+          }(walBatchingThreadPool)
+          f.onComplete { result =>
+              if (active) {
+                context.reply(result.get)
```

Comment:
It's better to log the error. Otherwise, it will just output the std err. See https://github.com/scala/scala/blob/a6e77f721921d9e2569433761a1e6a4056ae10f8/src/library/scala/concurrent/ExecutionContext.scala#L177

Actually, I suggest rewriting these codes like:
```Scala
          walBatchingThreadPool.execute(new Runnable {
            override def run(): Unit = Utils.tryLogNonFatalError{
              if (active) {
                context.reply(addBlock(receivedBlockInfo))
              } else {
                throw new IllegalStateException("Receiver Tracker Endpoint shut down.")
              }
            }
          })
```
This is much simpler